src/plugins/auto/plugin_config.cpp

```diff
@@ -0,0 +1,115 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include "utils/plugin_config.hpp"
+
+namespace MultiDevicePlugin {
+const std::set<std::string> PluginConfig::_availableDevices = {"AUTO", "CPU", "GPU", "TEMPLATE", "NVIDIA", "VPUX", "MULTI", "HETERO", "mock"};
+
+PluginConfig::PluginConfig() {
+    set_default();
+    device_property_validator = std::dynamic_pointer_cast<BaseValidator>(std::make_shared<FuncValidator>([](const ov::Any& target) -> bool {
+        auto deviceName = target.as<std::string>();
+        return _availableDevices.end() != std::find(_availableDevices.begin(), _availableDevices.end(),
+                                                            DeviceIDParser(deviceName).getDeviceName());
+    }));
+}
+
+void PluginConfig::set_default() {
+    register_property(
+        std::make_tuple(ov::enable_profiling, false),
+        std::make_tuple(ov::device::priorities, ""),
+        std::make_tuple(ov::hint::model_priority, ov::hint::Priority::MEDIUM),
+        std::make_tuple(ov::log::level, ov::log::Level::NO),
+        std::make_tuple(ov::intel_auto::device_bind_buffer, false),
+        std::make_tuple(ov::hint::performance_mode, ov::hint::PerformanceMode::UNDEFINED),
+        std::make_tuple(ov::hint::num_requests, 0, UnsignedTypeValidator()),
+        std::make_tuple(ov::intel_auto::enable_startup_fallback, true),
+        // TODO 1) cache_dir 2) allow_auto_batch 3) auto_batch_timeout
+        std::make_tuple(ov::cache_dir, ""),
+        std::make_tuple(ov::hint::allow_auto_batching, true),
+        std::make_tuple(ov::auto_batch_timeout, 1000),
+        // Legacy API properties
+        std::make_tuple(exclusive_asyc_requests, false),
+        // RO for register only
+        std::make_tuple(ov::device::full_name),
+        std::make_tuple(ov::device::capabilities),
+        std::make_tuple(ov::supported_properties));
+}
+void PluginConfig::register_property_impl(const ov::AnyMap::value_type& property, ov::PropertyMutability mutability, BaseValidator::Ptr validator) {
```

Comment:
Property definition itself has contained Mutability, why add additional Mutability parameter?

Reply:
original Property has been expanded to anyMap, and mutability here is for all the supported config/metric query